src/generic/complex_matrices.cc

```diff
@@ -538,6 +541,660 @@ namespace oomph
+
+    // Values from this matrix and matrix_in.
+    const std::complex<double>* i_values = this->value();
+    const std::complex<double>* in_values = matrix_in.value();
+
+    // The first entry in row_start is always zero.
+    res_row_start.push_back(0);
+
+    // Loop through the rows of both matrices and insert the column indices and
+    // values as a key-value pair.
+    for (unsigned row_i = 0; row_i < nrow_local; row_i++)
+    {
+      // Create the map for this row.
+      std::map<int, std::complex<double>> res_row_map;
+
+
+      int i_row_end = i_row_start[row_i + 1];
+      // Insert the column and value pair for this matrix.
+      for (int i = i_row_start[row_i]; i < i_row_end; i++)
+      {
+        res_row_map[i_column_indices[i]] = i_values[i];
+      }
+
+      int in_row_end = in_row_start[row_i + 1];
+      // Insert the column and value pair for in matrix.
+      for (int i = in_row_start[row_i]; i < in_row_end; i++)
+      {
+        res_row_map[in_column_indices[i]] += in_values[i];
+      }
+
+      // Fill in the row start
+      res_row_start.push_back(res_row_start.back() + res_row_map.size());
+
+      // Now insert the key into res_column_indices and value into res_values
+      for (std::map<int, std::complex<double>>::iterator it =
+             res_row_map.begin();
+           it != res_row_map.end();
+           ++it)
+      {
+        res_column_indices.push_back(it->first);
+        res_values.push_back(it->second);
+      }
+    }
+
+    // Finally build the result_matrix.
+    // Use the existing distribution.
+    result_matrix.build(res_values,
+                        res_column_indices,
+                        res_row_start,
+                        this->ncol(),
+                        this->nrow());
+  }
+
+  //=============================================================================
+  /// Element-wise addition of this matrix with matrix_in.
+  //=============================================================================
+  void CRComplexMatrix::add(const CRDoubleMatrix& matrix_in,
+                            CRComplexMatrix& result_matrix) const
+  {
+#ifdef PARANOID
+    // Check if this matrix is built.
+    if (!this->built())
+    {
+      std::ostringstream error_message;
+      error_message << "The matrix is not built.\n"
+                    << "Please build the matrix!\n";
+      throw OomphLibError(
+        error_message.str(), OOMPH_CURRENT_FUNCTION, OOMPH_EXCEPTION_LOCATION);
+    }
+
+    // Check if this matrix_in is built.
+    if (!matrix_in.built())
+    {
+      std::ostringstream error_message;
+      error_message << "The matrix matrix_in is not built.\n"
+                    << "Please build the matrix!\n";
+      throw OomphLibError(
+        error_message.str(), OOMPH_CURRENT_FUNCTION, OOMPH_EXCEPTION_LOCATION);
+    }
+
+    // Check if the dimensions of this matrix and matrix_in are the same.
+    unsigned long i_nrow = this->nrow();
```

Comment:
Did you possibly do a glob replace of `this_` with `i_`? It's possible this was `this_nrow`, but `i_nrow` looks quite odd and I'd expect something like `n_row`...